### PR TITLE
feat(dev): deploy operator to single namespace

### PR DESCRIPTION
**Pull request by aslakknutsen**
Add make target (un)deploy-operator-local to bind the operator to a single namespace.

OPERATOR_NAMESPACE=bookinfo make deploy-operator-local

Related #214

### CODE REVIEW

**Comment by aslakknutsen** [comment]
This generally works, but a few problems:

* only 'local' Operators can be installed on a single cluster, not combined with a Custer wide
* the deploy/undeploy install/uninstall the CRD as well so technically 1 can mess up for the others...

File: Makefile [hunk 1]
```diff
@@ -239,7 +240,8 @@ define process_template # params: template location
 		-p IKE_IMAGE_NAME=$(IKE_IMAGE_NAME) \
 		-p IKE_IMAGE_TAG=$(IKE_IMAGE_TAG) \
 		-p NAMESPACE=$(OPERATOR_NAMESPACE) \
-		-p TELEPRESENCE_VERSION=$(TELEPRESENCE_VERSION)
+		-p TELEPRESENCE_VERSION=$(TELEPRESENCE_VERSION) \
+    -p WATCH_NAMESPACE=$(OPERATOR_WATCH_NAMESPACE)
```
Comment (bartoszmajsak) [inline Makefile]:
```suggestion
		-p WATCH_NAMESPACE=$(OPERATOR_WATCH_NAMESPACE)
```

File: Makefile [hunk 2]
```diff
@@ -262,13 +264,35 @@ deploy-operator: ## Deploys istio-workspace operator resources to defined OPERAT
...
 	$(call process_template,deploy/istio-workspace/role_binding.yaml) | oc delete -n $(OPERATOR_NAMESPACE) -f -
 	oc delete -n $(OPERATOR_NAMESPACE) -f deploy/istio-workspace/role.yaml
 	oc delete -n $(OPERATOR_NAMESPACE) -f deploy/istio-workspace/service_account.yaml
 	oc delete -n $(OPERATOR_NAMESPACE) -f deploy/istio-workspace/crds/istio_v1alpha1_session_crd.yaml
 
+.PHONY: deploy-operator-local
+deploy-operator-local: export OPERATOR_WATCH_NAMESPACE=$(OPERATOR_NAMESPACE)
+deploy-operator-local: ## Deploys istio-workspace operator resources to a single Namespace defined by OPERATOR_NAMESPACE
+	$(call header,"Deploying local operator to $(OPERATOR_NAMESPACE)")
+	export OPERATOR_WATCH_NAMESPACE=$(OPERATOR_NAMESPACE)
```
Comment (bartoszmajsak) [inline Makefile]: Any reason why we have this here and in `275`?

File: deploy/istio-workspace/operator.yaml [hunk 3]
```diff
@@ -17,6 +17,10 @@ parameters:
     description: "The tag of the image to be used"
     required: true
     value: latest
+  - name: WATCH_NAMESPACE
+    description: "The tag of the image to be used"
```
Comment (bartoszmajsak) [inline deploy/istio-workspace/operator.yaml]: nene

Comment (aslakknutsen) [reply deploy/istio-workspace/operator.yaml]: :D https://github.com/Maistra/istio-workspace/pull/215/commits/27fdf4fbbafa5534a5f30a05aeda390197b04209

File: Makefile (hunk 2 above)
Comment (aslakknutsen) [reply Makefile]: Oh, there are reasons... but none that are valid. https://github.com/Maistra/istio-workspace/pull/215/commits/649f78fbd08f09a429d47264d544de66ee8d4bce

**Comment by aslakknutsen** [comment]
> Did you try to use it with ike install-operator -n local_ns?

Nope, not yet. Was thinking of maybe adding a --local flag to the install-operator ?

> Should we think about having a e2e test for it (issue here is that we have global operator installed beforehand)?

I was thinking more we change the e2e to use this on PR runs

**Comment by aslakknutsen** [comment]
> Why is your build failing?

Because you've hardcoded the whole Deployment in a test ;)

**Comment by bartoszmajsak** [comment]
> > Why is your build failing?
> 
> Because you've hardcoded the whole Deployment in a test ;)

It's a GOLDEN DEPLOYMENT

**Pull request merged**